test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class PackageExtractorTests : IDisposable
+    {
+        [Fact]
+        void PackageExtractor_withContentXmlFile()
+        {
+            // Arrange
+            var packageStream = TestPackages.GetTestPackageWithContentXmlFile();
+            var root = GetTempDir();
+            var packageReader = new PackageReader(packageStream);
+            var packagePath = Path.Combine(root.FullName, "packageA.2.0.3");
+            
+            // Act
+            var files = PackageExtractor.ExtractPackageAsync(packageReader, 
+                                                             packageStream, 
+                                                             new PackagePathResolver(root.FullName), 
+                                                             null, 
+                                                             PackageSaveModes.Nupkg, 
+                                                             CancellationToken.None).Result;
+            // Assert
+            Assert.False(files.Contains(Path.Combine(packagePath + "[Content_Types].xml")));
+            var test = Path.Combine(packagePath, "content/[Content_Types].xml");
+            Assert.True(files.Contains(Path.Combine(packagePath,"content/[Content_Types].xml")));
+        }
+
+        private DirectoryInfo GetTempDir()
+        {
+            var workingDir = new DirectoryInfo(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + "/"));
```

Comment:
Can you please use this TestFilesystemUtility.CreateRandomTestFolder()? And, then, simply call TestFilesystemUtility.DeleteRandomTestFolder? IMO, I would do it from the test itself to avoid a concurrent dictionary. However, if you called it from the dispose method, that is okay too.

Please remove this private method. This method here and other classes with similar code create a lot of stale directories since Path.GetTempPath() itself will create a folder